src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerInfo.cs

```diff
@@ -0,0 +1,33 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement
+{
+    /// <summary>
+    /// A class for storing various properties needed for Saving & Scripting a server
+    /// </summary>
+    public class ServerInfo : SqlObject
+    {
+        public string? HardwareGeneration { get; set; }
```

Comment:
Are these really all going to be possibly null? We should only use nullable where it's expected to have a possibly null value.

(if you're doing this because of the nullable warnings - it's fine to disable that for contract files like this)